oscarapi/basket/operations.py

```diff
@@ -66,12 +67,20 @@ def get_basket(request, prepare=True):
     return prepare_basket(basket, request) if prepare else basket
 
 
+def get_line_with_discounts(basket, line):
+    "Returns the in memory instance of a basket line with discounts applied"
+    # return the original line when it's not found in the cache
+    return next(x for x in basket.all_lines() if x.id == line.id)
```

Comment:
misschien voor de zekerheid next(x for x in basket.all_lines() if x.id == line.id, None) voor het geval all_lines() lege queryset returned?